sparse_list.py

```diff
@@ -55,12 +55,14 @@ def __setitem__(self, index, value):
     def __getitem__(self, index):
         try:
             s = slice(index.start, index.stop, index.step).indices(self.size)
-            return SparseList(
+            sl = SparseList(
                 {
                     k: self.elements[i]
                     for k, i in enumerate(xrange(*s))
                     if i in self.elements
                 }, self.default)
+            sl.size = len(xrange(*s))
```

Comment:
It feels like we should be able to calculate `sl.size` in constant time using a function of `index.start, index.stop & index.step`. See https://stackoverflow.com/a/36188683/78845 for example.

Reply:
Yes that's probably possible, but it's also likely that we would miss some edge cases…

Reply:
Actually I am wondering if Python does not already do the computation of``len(range(*s))` in constant time, given https://stackoverflow.com/a/36188948/985087

Reply:
Yes it seems that this is already done in constant time by Python, without enumerating all the indices. (Confirmed by testing `timeit`).

Reply:
```suggestion
            sl.size = len(indexes)
```